test/integration/good/code-gen/cl.expected

```diff
@@ -579,60 +579,64 @@ class optimize_glm_model final : public model_base_crtp<optimize_glm_model> {
       stan::math::fill(alpha_v, DUMMY_VAR__);
       
       current_statement__ = 1;
-      alpha_v = in__.vector(k);
+      assign(alpha_v, nil_index_list(),
+        in__.read<Eigen::Matrix<local_scalar_t__, -1, 1>, jacobian__>(lp__,
+          k), "assigning variable alpha_v");
```

Comment:
So I can change this if it's not generic enough but this line

```
in__.read<Eigen::Matrix<local_scalar_t__, -1, 1>, jacobian__>(lp__, k);
```

Should just be

```
in__.read<Eigen::Matrix<local_scalar_t__, -1, 1>>(k);
```

As jacobian calcs are only needed when calling constrains I made the read methods not take in the jacobian template value or lp.


Also we can shorten all of this up to remove the fill etc. for parameters and just have

```cpp
      Eigen::Matrix<local_scalar_t__, -1, 1> alpha_v = in__.read<Eigen::Matrix<local_scalar_t__, -1, 1>>(k);
```

Reply:
Ah I missed that thanks, shouldn't be hard.

Yeah, we should get rid of the fills, but I think that'll be a separate solution

Reply:
Yeah that could be a different matter. I think can check for `Decl` and do something with that